### ISSUE TITLE
Linting tftest.hcl files

### ISSUE THREAD

**Issue by wyardley** [terraform-linters/tflint#2128]
### Summary

It sounds from #2030 like `tftest.hcl` files should be able to be linted.

It seems, however, as if tflint may not be parsing `*.tftest.hcl` files (it should also recurse into subdirectories when in recursive mode if there are directories with no `.tf` files but with `.tftest.hcl` files).

Taking a look at the code, including [here](https://github.com/terraform-linters/tflint/blob/084b7d41ced66073c2475ff1cafee007a1438f8f/terraform/parser.go#L318-L324),  seems like some adjustments and tests might be needed for this to work?

### Command

tflint

### Terraform Configuration

```terraform
provider "google" {
  project = "foo-testproject"
}
variables {
  name    = "test-me"
}
run "basic" {
  command = plan
  plan_options {
    refresh = false
  }

  assert {
    condition     = google_storage_bucket.foo.bar.0.main_page_suffix == "index.html"
    error_message = "Incorrect website main page suffix."
  }
}
```

### TFLint Configuration

```HCL
plugin "google" {
  enabled = true
  version = "0.30.0"
  source  = "github.com/terraform-linters/tflint-ruleset-google"
}
```
```


### Output

```Shell
16:50:46 config.go:175: [INFO] Load config: .tflint.hcl
16:50:46 config.go:183: [INFO] file not found
16:50:46 config.go:190: [INFO] Load config: /Users/xx/.tflint.hcl
16:50:46 config.go:198: [INFO] file not found
16:50:46 config.go:201: [INFO] Use default config
16:50:46 config.go:476: [INFO] The "terraform" plugin block is not found. Enable the plugin "terraform" automatically
16:50:46 option.go:91: [DEBUG] CLI Options
16:50:46 option.go:92: [DEBUG]   CallModuleType: local
16:50:46 option.go:93: [DEBUG]   Force: false
16:50:46 option.go:94: [DEBUG]   Format: 
16:50:46 option.go:95: [DEBUG]   Varfiles: 
16:50:46 option.go:96: [DEBUG]   Variables: 
16:50:46 option.go:97: [DEBUG]   EnableRules: 
16:50:46 option.go:98: [DEBUG]   DisableRules: 
16:50:46 option.go:99: [DEBUG]   Only: 
16:50:46 option.go:100: [DEBUG]   EnablePlugins: 
16:50:46 option.go:101: [DEBUG]   IgnoreModules:
16:50:46 loader.go:39: [INFO] Initialize new loader
16:50:46 loader.go:81: [INFO] Building the root module while calling local child modules...
16:50:46 runner.go:46: [INFO] Initialize new runner for root
16:50:46 discovery.go:33: [INFO] Plugin "terraform" is not installed, but the bundled plugin is available.
16:50:46 discovery.go:54: [INFO] Plugin "terraform" found
16:50:46 [DEBUG] cmdrunner/cmd_runner.go:73: starting plugin: path=/opt/homebrew/bin/tflint args=["/opt/homebrew/bin/tflint", "--act-as-bundled-plugin"]
16:50:46 [DEBUG] cmdrunner/cmd_runner.go:80: plugin started: path=/opt/homebrew/bin/tflint pid=79171
16:50:46 [DEBUG] go-plugin@v1.6.1/client.go:827: waiting for RPC address: plugin=/opt/homebrew/bin/tflint
16:50:46 [DEBUG] go-plugin@v1.6.1/client.go:880: using plugin: version=11
16:50:46 [DEBUG] go-plugin@v1.6.1/client.go:1216: tflint: 16:50:46 [DEBUG] go-plugin@v1.6.1/server.go:419: plugin address: network=unix address=/var/folders/0z/yrnglm6n1l5_q19dv1yy0w3c0000gn/T/plugin2999297593
16:50:46 [DEBUG] host2plugin/client.go:124: starting host-side gRPC server
16:50:46 [DEBUG] go-plugin@v1.6.1/grpc_stdio.go:142: stdio: received EOF, stopping recv loop: err="rpc error: code = Unavailable desc = error reading from server: EOF"
16:50:46 [INFO]  go-plugin@v1.6.1/client.go:780: plugin process exited: plugin=/opt/homebrew/bin/tflint id=79171
16:50:46 [DEBUG] go-plugin@v1.6.1/client.go:558: plugin exited
```

### TFLint Version

0.53.0

### Terraform Version

v1.9.5

### Operating System

- [ ] Linux
- [X] macOS
- [ ] Windows

**Comment by wata727** [comment]
Thank you for raising this feature request.

We probably can't just use the current evaluator and parser as is because the [test language](https://developer.hashicorp.com/terraform/language/tests) has different semantics than the Terraform language. We would need to implement a dedicated emulator stack for the test language, which would require a fairly large refactoring.

See also https://github.com/terraform-linters/tflint/issues/1949

**Comment by wyardley** [comment]
A lot of the things can be parsed as-is, since the assertions are used in things like validation conditions. Even though there are some new fields possibly, I get at least some value (list index style for example) simply by renaming the files, and don’t notice any errors from the attributes that maybe `tflint` doesn’t know about.
